packages/builder/src/stores/portal/licensing.js

```diff
@@ -198,11 +198,11 @@ export const createLicensingStore = () => {
         const userQuota = license.quotas.usage.static.users
         const userLimit = userQuota?.value
         const userCount = usage.usageQuota.users
-        const userLimitReached = willReachUserLimit(userCount, userLimit)
-        const userLimitExceeded = willExceedUserLimit(userCount, userLimit)
+        const userLimitReached = usersLimitReached(userCount, userLimit)
+        const userLimitExceeded = usersLimitExceeded(userCount, userLimit)
         const days = dayjs(userQuota?.startDate).diff(dayjs(), "day")
         const userLimitDays = days > 1 ? `${days} days` : "1 day"
-        const warnUserLimit = userQuota?.startDate && userLimitExceeded
+        const errUserLimit = userQuota?.startDate && userLimitExceeded
```

Comment:
Some doubts about `warnUserLimit` var:

1. I've assumed that the proper name should be now `errUserLimit`; it is not a warning anymore; if limit is exceeded, account is directly locked. Rigth?
2. What's the point to check `userQuota?.startDate`? Do we still needing that? Why?

Reply:
1. Makes sense
2. We only set the `startDate` in certain circumstances:

https://github.com/Budibase/account-portal/blob/87c1dbe29c6a1f49a62ab2c472c557a1a76e58da/packages/server/src/sdk/licensing/quotas.ts#L295-L298
```ts
    account.hosting === Hosting.CLOUD &&
    plan.model === PlanModel.PER_USER && 
    quotas.usage.static.users.value !== UNLIMITED &&
    env.USER_LIMIT_START_DATE_ISO
 ```
 i.e.
- only for cloud
   - never shows in self host 
- only for per user plans 
   - never shows for people on existing plans that use day pass pricing 
- only when it's not unlimited
   - never show for people on unlimited users 

The start date acts as a whitelist flag for us here, basically saying 'the quota is only applicable under these circumstances'. In hindsight it could have been split into two fields, one for the actual date and another for if the date is applicable to the current user. But just negating the date altogether seemed like a simple way to achieve this. 

I think it would be nice if we could update the code to now be date aware as well, so that we can deploy prior to the date being passed and have the UI automatically react, rather than us being locked down to deploying on that date. 

if we can update to this 

```ts
const warnUserLimit = userQuota?.startDate && userLimitExceeded
const errUserLimit = warnUserLimit && Date.now() >= new Date(userQuota.startDate).getTime()
```

Then we can safely use the `errUserLimit` for showing the new modal / messaging, while keeping in the warnings in the current code. 

We can then either
- Target this PR at master, and bypass the main release process so that our current date of 1st june is correct, or
- Target at develop instead, and update account portal to change the date to 5th june, when the release is going out

Reply:
Alternatively we can keep as is, as I see you've already updated the warning banners. Doesn't make that much sense to keep in the old code given it will become immediately redundant when the date passes and we'll have to remove it. 

We'll just need to update the `USER_LIMIT_START_DATE_ISO` env var in account portal to reflect the next release date instead of june 1st

As for the `startDate`, we could either
1) Keep checking against the `startDate` flag
2) Move the logic required for showing the banner completely to the UI - i.e. 
   - cloud
   - user limit is not `-1` (unlimited)
   - plan.model == per user

I think my preference is #2 as it's really a frontend concern, we don't use the `startDate` on the backend

Reply:
From my point of view the usage of `startDate` as a flag to decide if a user is in a whitelist is confusing, because it is not an explicit behaviour; you just need to know it. Is very difficult to infer this by reading the code. On the other hand I wouldn't change this now; `startDate` is being used to calculate the remaining days for the user so we already have that value in the file; we can just use it for the moment. Maybe in a separated issue, we can make a little refactor to treat:

1. Make explicit that check (instead directly use startDate).
2. Move the responsibility to the front.

I can create a backlog issue and just move forward with this one. What do you think @Rory-Powell ?